### DISCUSSION TITLE
InnerSource Adoption Trends

### DISCUSSION THREAD

**Discussion by claredillon** [InnerSourceCommons/InnerSourceMarketing#268] [Show and tell]
The purpose of this thread is to keep track of reports of InnerSource adoption, emphasizing the growth in InnerSource momentum in previous years. It's intended for use by anyone creating resources regarding InnerSource and includes ISC and external references. If you come across something that could be of use, please do add it as a comment. Thanks!

## InnerSource Trends

Although the term was coined by Tim O’Reilly in the early 2000s, InnerSource began to gain momentum in 2015, coinciding with the foundation of [InnerSource Commons](https://innersourcecommons.org/). 

Since 2020, there has been another positive shift in momentum around InnerSource. In recent years, many industry research reports and opinion pieces have noted InnerSource as an important emerging software developer trends:
- InnerSource Commons [tracks 100+ public reports](https://innersourcecommons.org/stories/) of InnerSource practices.
- Perforce State of Open Source 2026 reports 16.5% of respondents have InnerSource projects. [2026 State of Open Source Report](https://www.perforce.com/resources/opl/2026-state-of-open-source-report).
- Stack Overview reports 16% of respondents (5.5K+) practice InnerSource in [their 2022 survey](https://survey.stackoverflow.co/2022/#section-developer-experience-developer-experience-processes-tools-and-programs-within-an-organization). In their [2023 StackOverview Survey](https://survey.stackoverflow.co/2023/#section-developer-experience-developer-experience-processes-tools-and-programs-within-an-organization), the percentage went down to 14.13% - but that corresponded to a higher number of respondents than 2022 saying they practiced InnerSoruce (5.9K vs 5.5K). 
- Gartner includes InnerSource on the Software Engineering hype cycle in [2022](https://x.com/TheMarkONeill/status/1557531379746881536?s=20), [2023](https://www.linkedin.com/feed/update/urn:li:activity:7132928587778682880?updateEntityUrn=urn%3Ali%3Afs_feedUpdate%3A%28V2%2Curn%3Ali%3Aactivity%3A7132928587778682880%29), [2024](https://www.harness.io/blog/platform-engineering-beyond-the-trough-of-disillusionment), and [2025](https://www.youtube.com/watch?v=iKhIXezMeys)
- Gartner lists InnerSource as a [top strategic technology trend in software engineering in 2023 and beyond](https://www.gartner.com/en/newsroom/press-releases/gartner-identifies-the-top-strategic-technology-trends-in-software-engineering-trends-for-2023).
- Linux Foundation reports at Open Source Finance Forum 2023 that over half of the respondents to the [2023 State of Open Source in Financial Services research](https://project.linuxfoundation.org/hubfs/LF%20Research/FINOS%20Annual%20Report%202022%20-%20Report.pdf?hsLang=en) contribute to InnerSource each week.

![image](https://github.com/InnerSourceCommons/InnerSourceMarketing/assets/32961070/4f283b8b-faca-4c0e-ac06-6c9bc6d0ac6c)

**Comment by claredillon** [comment]
In [2023 StackOverview Survey](https://survey.stackoverflow.co/2023/#section-developer-experience-developer-experience-processes-tools-and-programs-within-an-organization), the percentage went down to 14.13% - but that corresponded to a higher number of respondents than 2022 saying they practiced InnerSoruce (5.9K vs 5.5K).

<img width="763" alt="image" src="https://github.com/InnerSourceCommons/InnerSourceMarketing/assets/32961070/f1e3dfde-2261-441d-99de-2bee194b6f1e">

**Comment by claredillon** [comment]
33% of respondents to the Linux Foundation 2023 State of OSPOs and OSS Initiatives list InnerSource as a core responsibility.

<img width="737" alt="image" src="https://github.com/InnerSourceCommons/InnerSourceMarketing/assets/32961070/bb2ee5af-a74d-4d55-afc7-8f43f9207159">

**Reply by claredillon** [reply]
The same study includes this on InnerSource, exploring how InnerSource initiatives relate to multi-stakeholder involvement in InnerSource initiatives.

<img width="624" alt="image" src="https://github.com/InnerSourceCommons/InnerSourceMarketing/assets/32961070/124b22e1-aae2-4d93-8267-24145c516015">

**Comment by claredillon** [comment]
Copying references for InnerSource -> Open Source examples from Slack discussion here: 
dirkriehle
  Jun 30th at 11:45 AM
Hi everyone, can you point us to open source projects that used to be inner source projects? For our data-driven analysis incl. transfer pricing. Thank you!




9 replies


mainec
  [Jun 30th at 3:29 PM](https://innersourcecommons.slack.com/archives/C04PXKRN4/p1751293741532639?thread_ts=1751280311.350589&cid=C04PXKRN4)
[@Cornelius Schumacher](https://innersourcecommons.slack.com/team/UMZ4YHJF6)
 
[@Carsten Hoffmann](https://innersourcecommons.slack.com/team/U08DS220BT3)
 anything from your organisation, or is my brain mixing things up here?


Clare Dillon
  [Jun 30th at 3:46 PM](https://innersourcecommons.slack.com/archives/C04PXKRN4/p1751294810674259?thread_ts=1751280311.350589&cid=C04PXKRN4)
My favourite example is Backstage from Spotify. Talked about in their ISC talk which should be on the YouTube channel.
:innersource-is-coming:
1

[3:47](https://innersourcecommons.slack.com/archives/C04PXKRN4/p1751294871961259?thread_ts=1751280311.350589&cid=C04PXKRN4)
And I believe Porche's Design system was originally InnerSource before open sourced...
[3:49](https://innersourcecommons.slack.com/archives/C04PXKRN4/p1751294982283149?thread_ts=1751280311.350589&cid=C04PXKRN4)
And I do remember one community call where someone talked about the fact that they switched from InnerSource to open source just to get around issues like transfer pricing - but I can't remember the source, sorry.


russ
  [Jun 30th at 5:07 PM](https://innersourcecommons.slack.com/archives/C04PXKRN4/p1751299671855159?thread_ts=1751280311.350589&cid=C04PXKRN4)
[@JerryTan](https://innersourcecommons.slack.com/team/U0162MT8P7A)
 shared that [Apache Fory](https://github.com/apache/fory) started as InnerSource.

apache/fory
A blazingly fast multi-language serialization framework powered by JIT and zero-copy.
Website
https://fory.apache.org/
Stars
3326
Added by [GitHub](https://innersourcecommons.slack.com/services/B026P4J0EHX)
:heart:
1



dirkriehle
  [Jul 1st at 7:25 AM](https://innersourcecommons.slack.com/archives/C04PXKRN4/p1751351145990279?thread_ts=1751280311.350589&cid=C04PXKRN4)
Thanks, 
[@Clare Dillon](https://innersourcecommons.slack.com/team/UQCNRUS4V)
 and 
[@russ](https://innersourcecommons.slack.com/team/U4S1WTCHG)
!


gilyehuda
  [Friday at 11:26 PM](https://innersourcecommons.slack.com/archives/C04PXKRN4/p1751667976025349?thread_ts=1751280311.350589&cid=C04PXKRN4)
Apache Pulsar was an internal project at Yahoo that I wanted to make InnerSource but the corresponding team did not want to participate. So we published it as an open source project so that the internal teams would be more willing to use it and contribute to it.
:+1:
1
:heart:
2



Guilherme Dellagustin
  [Yesterday at 1:51 PM](https://innersourcecommons.slack.com/archives/C04PXKRN4/p1752065514782749?thread_ts=1751280311.350589&cid=C04PXKRN4)
If I'm not mistaken, from microsoft, VS code started InnerSource.
From SAP, I can list from the top of my head:
https://github.com/SAP/abap-cleaner
https://github.com/openmcp-project (parts of it at least, I don't get the best overview).
I'm sure there are many more, but I'd have to dig deeper.
:heart:
2



Tom Sadler
  [Yesterday at 2:12 PM](https://innersourcecommons.slack.com/archives/C04PXKRN4/p1752066724042149?thread_ts=1751280311.350589&cid=C04PXKRN4)
Couple of BBC examples:
https://github.com/bbc/bigscreen-player
https://github.com/bbc/morty-docs
In both examples I believe we created a fresh initial git commit as there are data and privacy risks to opening up previously private git history.

**Comment by claredillon** [comment]
Reference 2024 Gartner Hype Cycle for Software Engineering - taken from here: https://www.harness.io/blog/platform-engineering-beyond-the-trough-of-disillusionment 
<img width="905" height="524" alt="image" src="https://github.com/user-attachments/assets/a0fc4741-7292-4d7c-adc3-a522d9985321" />

**Reply by lenucksi** [reply]
Interesting. Is there a more recent version, e.g. 2025 available? Supply Chain Security is seeing EU funding, especially given CRA regulations etc. Thus that might have moved into the enlightenment already.

**Reply by claredillon** [reply]
They don't make them available without payment - so we're relying on others publishing them. We should only reference them where others do (not show the hype cycle ourselves)

**Comment by claredillon** [comment]
Gartner Blog Post in Sep 2024
https://www.gartner.com/en/documents/5806215

Maximize Platform Development Velocity Using InnerSource
Published: 30 September 2024

Summary
Digital product managers face increasing pressure to deliver innovation quickly into their shared platforms. To this end, DPMs can use InnerSource to fast-track development and leverage resources from across the portfolio.

**Reply by lenucksi** [reply]
Looks like it needs a subscription, i.e. paywall...

**Reply by claredillon** [reply]
Yes - we don't have access to the article - but noting the reference by Gartner and the summary: Summary
Digital product managers face increasing pressure to deliver innovation quickly into their shared platforms. To this end, DPMs can use InnerSource to fast-track development and leverage resources from across the portfolio.

**Comment by claredillon** [comment]
Reference to Gartner's 2024 Hype cycle for Software Engineering. InnerSource is also on the 2025 Gartner Hype Cycle (as confirmed to us by email from Gartner on 4th Nov 2025) - moving into the trough of disillusionment.

https://www.linkedin.com/posts/markwoneill_is-platform-engineering-falling-into-the-activity-7297972628084936704-chYL/

**Comment by claredillon** [comment]
Reference: InnerSource on Gartner Hype Cycle for Software Engineering 2025

<img width="624" height="347" alt="image" src="https://github.com/user-attachments/assets/ec0fd61b-4836-4971-bc99-33760d149916" />

Presented at the InnerSource Summit 2025, see more here: 

https://www.youtube.com/watch?v=iKhIXezMeys

**Reply by claredillon** [reply]
This presentation references a decrease in InnerSource practice. 

<img width="623" height="344" alt="image" src="https://github.com/user-attachments/assets/085c8ff8-b18c-4c47-8afa-a501645edb57" />

Please note this may be where InnerSource is seen to be a step on the path to open source. Implies we need to help folks recognize that InnerSource is more than this!

<img width="624" height="347" alt="image" src="https://github.com/user-attachments/assets/c280f813-0dbb-484c-86b0-f83cf6d106db" />

**Comment by claredillon** [comment]
State of Open Source 2026 by Perforce, from presentation on May 7th 2026

<img width="1493" height="849" alt="image" src="https://github.com/user-attachments/assets/03e967c2-764f-459c-aa2a-76375946663c" />

**Reply by claredillon** [reply]
It was noted in the presentation about the report that a lot of respondents came from smaller orgs.
